### PR TITLE
add preference to run always (ignore run conditions)

### PR DESCRIPTION
**Pull request by fps**
As I wrote in the comment https://github.com/syncthing/syncthing-android/pull/1821#issuecomment-1213453291 for https://github.com/syncthing/syncthing-android/pull/1821 it might be useful to have a preference to ignore all run conditions. This PR implements this. Tested in the emulator and seems to work
run_conditions_enabled
run_conditions_disabled
.

### CODE REVIEW

File: app/src/main/res/values/strings.xml
Comment (imsodin) [inline app/src/main/res/values/strings.xml:322]: How does this give manual control? It means it runs all the time, no control whatsover, right? Seems like just the first sentence would cut it.

**Review by imsodin — COMMENTED** [review]
Could you please disable (grey out) all the other settings when this one is active to make it visually clear it overrides everything.

File: app/src/main/res/values/strings.xml
Comment (fps) [reply app/src/main/res/values/strings.xml:322]: The manual control is then exercised by starting and exiting the app. I agree that this option is badly worded all together. I guess the clearest labeling would be "Enable Run Conditions" as title with the default being true (to keep the behavior from before this PR unless explicitly changed). If set to false all run condition options are greyed out (like you mentioned in your other comment) and the run conditions are ignored What do you think? Thanks for the review!

Comment (imsodin) [reply app/src/main/res/values/strings.xml:322]: makes sense

**Comment by fps** [comment]
Hi, I reworked the code a little bit to implement the previously discussed naming changes and the greying out of the other preferences.

I am not too happy with the duplication of the logic to enable/disable the WIFI related preferences. Also I'm not an adroid developer by any stretch so I don't know if the iteration over the preferences by index in the run condition category is the smartest thing to do there.. Happy to get more feedback on this.. Thanks!

**Comment by fps** [comment]
Oh, and I guess that's a little embarrassing: When I actually installed the apk on my device I noticed that the new version had working connectivity detection even with wireguard enabled (and with run conditions enabled.) Possibly the f-droid version I had installed previously (v1.20.4) was just too old. I'll leave this PR open anyways because disabling run conditions might be useful for future problems..

**Comment by fps** [comment]
Actually: Ignore my previous comment. Sometimes the connectivity check just takes a while to kick in. So I actually still do need this preference to run syncthing with wireguard enabled. Sorry for the noise ;)

File: app/src/main/java/com/nutomic/syncthingandroid/activities/SettingsActivity.java
Comment (imsodin) [inline app/src/main/java/com/nutomic/syncthingandroid/activities/SettingsActivity.java:380]: Forgotten debug log?

**Review by imsodin — COMMENTED** [review]
Looks good. I assume you tested it and it works as expected? Screenshot for UI changes is always appreciated (though here it's pretty obvious).

File: app/src/main/java/com/nutomic/syncthingandroid/activities/SettingsActivity.java
Comment (fps) [reply app/src/main/java/com/nutomic/syncthingandroid/activities/SettingsActivity.java]: Yes, thanks. Removed that line. In my tests it works fine. Of course I'm a bit wary to actually have my amateurish code in such a widely used project without further testing by other people. So if anyone else wants to give this a spin they would make me very happy..

**Comment by fps** [comment]
I attached two screenshots as well. I'm not yet convinced this is the cleanest UI, so I'm open to suggestions. I'm not too well versed in the android UI guidelines (not at all) and I'm not too much of an android dev (or even user), so no offence taken if I missed an obviously better way to present this..

**Comment by tomasz1986** [comment]
What do you think about making the explanation a little clearer, e.g. by changing

```
<string name="run_conditions_summary">Use the following options to decide when Syncthing will run.</string>
```

to

```
<string name="run_conditions_summary">Use the following options to decide when Syncthing will run. When disabled, Syncthing will run all the time.</string>
```

?

File: app/src/main/java/com/nutomic/syncthingandroid/activities/SettingsActivity.java
Comment (imsodin) [reply app/src/main/java/com/nutomic/syncthingandroid/activities/SettingsActivity.java]: There's going to be a testing phase when the RC is released to the play beta channel. This seems safe enough (hey, the mainly active maintainer here isn't exactly an android specialist either ;) ).

**Comment by imsodin** [comment]
I personally don't like longer strings, I think that just means they are more likely to be ignored. And I don't think the second sentence adds any crucial new info. And it means translation effort. So I'd tend against doing the change, but no very strong opinion.

**Review by imsodin — APPROVED** [review]
lgtm. The branch needs a rebasing though.

**Comment by imsodin** [comment]
That makes no sense though: What's the purpose of something that is running but also not? Then again I am obviously biased as I know how it works. Anyway I'll defer to @fps or anyone else to chime in, I really don't care much.

**Comment by fps** [comment]
I see @tomasz1986's point of some more clarification needed. But I'm not sure on how to reword that to make it more clear. Your suggestion, @tomasz1986, can also be a source of confusion (as "always run" can be misunderstood to really "always run", i.e. there's no more way to turn syncthing off at all, which is not true either.) 

I'd leave it as is right.. Thanks for the suggestion though :)

**Review by imsodin — APPROVED** [review]

**Comment by tomasz1986** [comment]
The PR is already merged, but I'd like to ask one more question, as I can't really test this at the moment.

What happens when you upgrade from the older version of the app? Do the current settings persist (i.e. so that run conditions are enabled) or does it disable them?

**Comment by fps** [comment]
The new preference defaults to <code>true</code>, so the original behaviour is preserved on an upgrade.

**Pull request merged**
